frontend/catalyst/pennylane_extensions.py

```diff
@@ -935,9 +945,21 @@ def check_is_iterable(x, hint):
         fn = _ensure_differentiable(f)
         grad_params = _check_grad_params(method, scalar_out, h, argnum)
 
-        jaxpr, _ = _make_jaxpr_check_differentiable(fn, grad_params, *params)
+        jaxpr, out_tree = _make_jaxpr_check_differentiable(fn, grad_params, *params)
 
         results = vjp_p.bind(*params, *cotangents, jaxpr=jaxpr, fn=fn, grad_params=grad_params)
```

Comment:
Shouldn't we flatten the params and cotangents before binding?

Reply:
Here the cotangent must have the same shape as the returns, I considered that it was on the user side to give the cotangents with the right shape.

Reply:
Right they should match, so normally I think the user would match the PyTree shape for the results and contangents.

Reply:
There is also the issue of params which could be PyTrees

Reply:
Yes here I did not consider the params as pytrees.

Reply:
done in #549